closed/src/java.base/share/classes/jdk/crypto/jniprovider/NativeCrypto.java

```diff
@@ -119,14 +119,15 @@ public static final boolean isAllowedAndLoaded() {
 
     /**
      * Return the OpenSSL version.
-     * -1 is returned if CRIU is enabled and the checkpoint is allowed.
+     * -1 is returned if CRIU is enabled and checkpoints are allowed
+     * unless -XX:-CRIUSecProvider is specified.
      * The libraries are to be loaded for the first reference of InstanceHolder.instance.
      *
      * @return the OpenSSL library version if it is available
      */
     public static final long getVersionIfAvailable() {
 /*[IF CRIU_SUPPORT]*/
-        if (InternalCRIUSupport.isCheckpointAllowed()) {
+        if (InternalCRIUSupport.isCheckPointAllowed() && InternalCRIUSupport.enableCRIUSecProvider()) {
```

Comment:
typo: isCheckPointAllowed -> isCheckpointAllowed

At the very least you ought to ensure the code compiles before pushing to your fork:
```
support/j9jcl/java.base/share/classes/jdk/crypto/jniprovider/NativeCrypto.java:127: error: cannot find symbol
        if (InternalCRIUSupport.isCheckPointAllowed() && InternalCRIUSupport.enableCRIUSecProvider()) {
                               ^
  symbol:   method isCheckPointAllowed()
  location: class InternalCRIUSupport
```

Reply:
Updated